### ISSUE TITLE
Adding beams and early stopping to translation

### ISSUE THREAD

**Issue by xirect** [amaiya/ktrain#243]
Hi @amaiya!

I found something that increases translation speed and wanted to share that with you, it would also be really cool if you could integrate it within ktrain! When in need of faster translation with less accuracy, the number of beams can be lowered.

```
def translate_sentences(self, sentences, num_beams=4, early_stopping=True):
     """
    translate sentence using model_name as model
    Args:
      sentences(list): list of strings representing sentences that need to be translated
                     IMPORTANT NOTE: Sentences are joined together and fed to model as single batch.
                                     If the input text is very large (e.g., an entire book), you should
                                     break it up into reasonbly-sized chunks (e.g., pages, paragraphs, or sentences) and 
                                     feed each chunk separately into translate to avoid out-of-memory issues.
    Returns:
      str: translated sentences
    """
    import torch
    with torch.no_grad():
        translated = self.model.generate(**self.tokenizer.prepare_seq2seq_batch(sentences).to(self.torch_device), num_beams=num_beams, early_stopping=early_stopping)
        tgt_sentences = [self.tokenizer.decode(t, skip_special_tokens=True) for t in translated]
    return tgt_sentences****
```

I found this little gem in the huggingface documentation

```
from transformers import BartTokenizer, BartForConditionalGeneration, BartConfig

# see ``examples/summarization/bart/run_eval.py`` for a longer example
model = BartForConditionalGeneration.from_pretrained('facebook/bart-large-cnn')
tokenizer = BartTokenizer.from_pretrained('facebook/bart-large-cnn')

ARTICLE_TO_SUMMARIZE = "My friends are cool but they eat too many carbs."
inputs = tokenizer([ARTICLE_TO_SUMMARIZE], max_length=1024, return_tensors='pt')

# Generate Summary
summary_ids = model.generate(inputs['input_ids'], num_beams=4, max_length=5, early_stopping=True)
print([tokenizer.decode(g, skip_special_tokens=True, clean_up_tokenization_spaces=False) for g in summary_ids])
```

**Comment by xirect** [comment]
I created a PR #244

**Comment by xirect** [comment]
num_beams=1 is apparantly the default, might still be nice to be able to tweak the amount of beams.

**Comment by amaiya** [comment]
@xirect:  Thanks!  Great find.   I went ahead and already added in these parameters per your request **before** seeing that you already implemented a PR (sorry about that).   Thanks for the PR, though, as I reviewed it and also copied over the `half` parameter as you did.

I did notice that these parameter settings affect the output slightly but not necessarily in a bad way.  Nevertheless, I decided to leave the defaults to `None` to be consistent with what is in the `transformers` library (which means `transformers` will set `num_beams=1` and `early_stopping=False`).    So, when you use, you can explicitly set these parameters as appropriate to obtain the speedup.

Thanks.

**Comment by amaiya** [comment]
This is now available in v0.21.1.  Thanks again, @xirect
